datastore/postgres/migrations/migrations_test.go

```diff
@@ -0,0 +1,67 @@
+package migrations
+
+import (
+	"bufio"
+	iofs "io/fs"
+	"os"
+	"path"
+	"regexp"
+	"sort"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestMigrationsMismatch(t *testing.T) {
+	var migrations, files []string
+
+	// Get referenced migrations
+	migrationLine, err := regexp.Compile(`runFile\(\"(.*)\"\)`)
```

Comment:
This feels brittle. An idea can be to create a base slice like

```
const indexerMigrations = []string{
	"filename1",
	"filename2",
	...
}
```

and then have

```
func init() {
	for i, migration := range indexerMigrations {
		IndexerMigrations = append(IndexerMigrations, {
			ID: i+1,
			Up: runFile(migration),
		})
	}
}
```

something like that so we can just ensure `indexerMigrations` lists all the files

Reply:
I think one thing we'd want to keep is the ID being explicitly associated with the migration file so a user can check their DB migration version and know which was the last file being run (without having to work it out).

It is brittle in the sense that the runFile might change, but the proposed solution doesn't address fact that a sql file might exist on the FS but not be referenced in the go code, maybe this isn't actually an issue if you aren't forgetful (like myself) and remember to reference the migration. The reason this case was slightly different was that there was an admin command added to `clairctl` to pre-create the index so when testing the code in stage and production the index was already create and hence, it didn't look like anything was missing.